fossology/fossology
# feat(Installation Script): Enhance Cross-Platform Compatibility and Line Encoding Issue

## Description
This PR aims to update `fo-installdeps`, `install_composer` & `fo-cleanold` installation script to improve line encoding handling across windows. The goal is to ensure that these script function runs seamlessly across unix-like systems, [Windows Subsystem for Linux(WSL)](https://learn.microsoft.com/en-us/windows/wsl/), & other environments by managing line encoding & utilising MacOs, package manager, [Homebrew](https://brew.sh/) for installing project dependencies efficiently. 

### Changes

 - Implemented automatic detection and normalization of line endings using `dos2unix` to ensure scripts execute without errors across different platforms.
 - Added checks for the presence of necessary utilities (`curl`, `dos2unix`), with automated installation prompts if not found.
 - Adjusted script behavior based on user execution privileges, modifying default installation paths and permissions to suit different user levels.

## How to test
 - Ensure that the necessary utilities and configurations are correctly in place. Use the following commands to check the status of the installed services:
 
   #### For systemd-based systems:
         systemctl status fossology 
   #### For SysV-based systems:
        /etc/init.d/fossology status 

 - Confirm that the FOSSology Apache HTTPD configuration is properly loaded by the Apache server.Use this command to test the configuration:
   <pre> <code> apachectl configtest </code> </pre> 


CC:  [Gaurav Mishra](https://github.com/GMishx), [Kaushlendra Pratap](https://github.com/Kaushl2208)
